File: caesar.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <stdlib.h> 
#include <wctype.h>

int main(int argc, string argv[])
{
    long int key;
    int n;
    int i;
    string plaintext;
    
    if (argc == 2)
    {
        key = atoi(argv[1]);
        plaintext = get_string("plaintext: ");
        
        printf("ciphertext: ");
        
        n = strlen(plaintext);
        for (i = 0; i < n; i++)
        {
            if (iswlower(plaintext[i]) != 0)
            {
                printf("%c", (char)('a' + (plaintext[i] + key - 'a') % 26));
            } 
            else if (iswupper(plaintext[i]) != 0)
            {
                printf("%c", (char)('A' + (plaintext[i] + key - 'A') % 26));
            }
            else
            {
                printf("%c", plaintext[i]);
            }
        }
        
        printf("\n");
        
        return 0;
    }
    else
    {
        printf("Usage: ./caesar key");
        return 1;
    }
}

File: caesar.py
from cs50 import get_string
from sys import argv

if len(argv) == 2:
    key = argv[1]
    if not key.isdigit():
        print('1')
else:
    print('Usage: python caesar.py k')
    exit(1)

key = int(key)
plaintext = get_string('plaintext: ')
print('ciphertext: ', end='')

ai = ord('a')
Ai = ord('A')

for c in plaintext:
    ci = ord(c)

    if c.isalpha():
        if c.islower():
            print(chr(ai + (ci + key - ai) % 26), end='')
        else:
            print(chr(Ai + (ci + key - Ai) % 26), end='')
    else:
        print(c, end='')



